tools/dogfooding/read-dogfooded-hash.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# -----------------------------------------------------------
+# Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2019-Present Datadog, Inc.
+# -----------------------------------------------------------
+
+import sys
+import traceback
+import argparse
+from src.dogfood.package_resolved import PackageResolvedFile, PackageID
+from src.utils import print_err, print_info
+
+def print_to_stdout(hash):
+    """
+    Prints the hash in a format that is recognized by caller script.
+    """
+    print(f'DOGFOODED_HASH={hash}')
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Reads the hash of dd-sdk-ios dependency in "Package.resolved" of SDK-dependant project and prints it to STDOUT.')
+    parser.add_argument('--repo-package-resolved-path', type=str, required=True, help='Path to "Package.resolved" file in SDK-dependant project')
+    args = parser.parse_args()
+    
+    try:
+        dependant_package = PackageResolvedFile(path=args.repo_package_resolved_path)
```

Comment:
/question

Could we rely on `__dogfoodedSDKVersion` to keep it simple?

Reply:
Good idea, I like the concept of using `__dogfoodedSDKVersion` as source of truth here 👍. However, we use `dogfooded-hash` for later query in `git log` (changelog) and that needs full commit hash vs `__dogfoodedSDKVersion` only encodes the short one for readability. I will check what we can do here. 

Ideas @ganeshnj ?

Reply:
the short form is totally fine here, git handles it gracefully. You will almost never hit a conflict.

Reply:
Done ✅